### ISSUE TITLE
IPAC table output fails with masked values and column format using repr

### ISSUE THREAD

**Issue by stargaser** [astropy/astropy#8354]
My colleagues at IPAC have found a case in which writing a table in IPAC format with masked values and a format that includes `repr` in it, results in invalid input. The table that is output cannot be read back in.

Here is a code snippet to reproduce the problem:

```python
from io import StringIO
from astropy.table import Table, MaskedColumn

table = Table([MaskedColumn(data=[1.0, 2.0, 3.0],
                          mask=[False, True, False],
                          dtype='f8', format='{!r:>}')])
out = StringIO()
table.write(out, format='ipac')
str_vals = out.getvalue()

print(str_vals)
```

This prints
```
|  col0|
|double|
|      |
|'null'|
    1.0 
   null 
    3.0
```

Trying to read the values with `Table.read(str_vals, format='ipac')` raises `ValueError: Column col0 failed to convert: could not convert string to float: 'null'`, because the null string doesn't match the null values in the column.

If the value of the column format is `None`, or is instead `'{!s:>}'`, the problem doesn't happen.

I _think_ this is a problem to be fixed in astropy.io.ascii, but first let me explain that the case of using `format='{!r:>}` arises in astropy.io.votable. When precision and width are not provided in a VOTable, the format that uses `repr` is inserted. For example, a user querying WISE catalogs using astroquery and writing the output to an IPAC table will encounter this problem, because a VOTable is obtained from the IRSA archive and then read in with astropy.io.votable. I found #5432 introduced this use of `repr` in the column format in astropy.io.votable.

I have a code fix but I first want to see if there is agreement that this should be fixed in the IPAC table writer and not on astropy.io.votable.

cc @bsipocz @pllim @vandesai1 

UPDATE: Per the discussion below, it has become clear that this example is contrived, and I believe it is highly unlikely that a user would use a `repr`-style format with IPAC tables, on their own. The real problem arises when a VOTable from astroquery is converted to Table, as shown in [my comment](https://github.com/astropy/astropy/issues/8354#issuecomment-454826394) below.

EDIT: Syntax highlighting
EDIT #2: add notes that the example above is contrived.

**Comment by pllim** [comment]
cc @taldcroft

**Comment by taldcroft** [comment]
Well that's pretty funky.  :smile:  If you have a fix for this in `io.ascii` that would be worth having a look.

**Comment by stargaser** [comment]
Here is a short snippet showing how this happens when a user queries WISE with astroquery and writes the output in IPAC table format.

```
from astroquery.irsa import Irsa

table = Irsa.query_region("10h45m +59d", catalog="allwise_p3as_psd", spatial="Cone",
    selcols="ra,dec,w4mpro,w4sigmpro")
table.write("test_nulls.tbl", format="ipac")
```

This is the content of `test_nulls.tbl`:
```
|     ra|    dec|        clon|        clat|w4mpro|w4sigmpro|    dist|     angle|  id|
| double| double|        char|        char|double|   double|  double|    double|char|
|    deg|    deg|            |            |   mag|      mag|    arcs|       deg|    |
|   null|   null|        null|        null|'null'|   'null'|  'null'|    'null'|null|
 161.248  59.002 10h44m59.49s 59d00m07.17s   8.87      null 8.197288 331.004533    0
```

**Comment by taldcroft** [comment]
I spent a little time understanding this better and now I might advocate an upstream fix in votable.  Basically the repr format string that votable uses is (as far as I can tell) so that VOtable can output data appropriately in full precision.  However, astropy Table does full-precision output by default already if `format` is not defined (None).  That convention is pretty much assumed throughout Table and `io.ascii`.

So I think the right answer here is to fix how VOtable gets converted to Table, namely block the votable-specific repr format convention from getting into Table:
```
diff --git a/astropy/io/votable/tree.py b/astropy/io/votable/tree.py
index 07035fcb7..304e06f4c 100644
--- a/astropy/io/votable/tree.py
+++ b/astropy/io/votable/tree.py
@@ -1530,7 +1530,8 @@ class Field(SimpleElement, _IDProperty, _NameProperty, _XtypeProperty,
         if self.unit is not None:
             # TODO: Use units framework when it's available
             column.unit = self.unit
-        if isinstance(self.converter, converters.FloatingPoint):
+        if (isinstance(self.converter, converters.FloatingPoint) and
+                self.converter.output_format != '{!r:>}'):
             column.format = self.converter.output_format 
```
This fixes the original issue and might fix other latent problems with io.ascii and VOTables.

**Comment by stargaser** [comment]
This sounds good to me. I have updated #8358 with this fix. I think I need some advice on how to write a test case that doesn't use astroquery.

**Comment by pllim** [comment]
Fixed in #8358
